hazelcast/src/main/java/com/hazelcast/map/impl/eviction/EvictionOperator.java

```diff
@@ -16,239 +16,26 @@
-
-            boolean expired = recordStore.isExpired(record, now, false);
-            if (expired) {
-                // Fire EXPIRED event also because there is a possibility that
-                // evicted entry may be also an expired one.
-                fireEvent(key, value, mapName, EXPIRED, mapServiceContext);
-            }
-            fireEvent(key, value, mapName, EVICTED, mapServiceContext);
-        }
-
-        return true;
-    }
-
-    private long[] createAndPopulateEvictionCriteriaArray(RecordStore recordStore,
-                                                          EvictionPolicy evictionPolicy) {
-        final int size = recordStore.size();
-        long[] criterias = null;
-        int index = 0;
-        final Iterator<Record> iterator = recordStore.iterator();
-        while (iterator.hasNext()) {
-            final Record record = iterator.next();
-            if (criterias == null) {
-                criterias = new long[size];
-            }
-            criterias[index] = getEvictionCriteriaValue(record, evictionPolicy);
-            index++;
-            //in case size may change (increase or decrease) when iterating.
-            if (index == size) {
-                break;
-            }
-        }
-        if (criterias == null) {
-            return null;
-        }
-        // just in case there may be unassigned indexes in criterias array due to size variances
-        // assign them to Long.MAX_VALUE so when sorting asc they will locate
-        // in the upper array indexes and we wont care about them.
-        if (index < criterias.length) {
-            for (int i = index; i < criterias.length; i++) {
-                criterias[i] = Long.MAX_VALUE;
-            }
-        }
-        return criterias;
-    }
-
-    private int getEvictionStartIndex(long[] criterias, int evictableSize) {
-        final int length = criterias.length;
-        final int sizeToEvict = Math.min(evictableSize, length);
-        final int index = sizeToEvict - 1;
-        return index < 0 ? 0 : index;
-    }
-
-    public void fireEvent(Data key, Object value, String mapName, EntryEventType eventType,
-                          MapServiceContext mapServiceContext) {
-        if (!hasListener(mapName)) {
-            return;
-        }
-        final MapEventPublisher mapEventPublisher = mapServiceContext.getMapEventPublisher();
-        final NodeEngine nodeEngine = mapServiceContext.getNodeEngine();
-        final Address thisAddress = nodeEngine.getThisAddress();
-        final Data dataValue = mapServiceContext.toData(value);
-        mapEventPublisher.publishEvent(thisAddress, mapName, eventType, true,
-                key, dataValue, null);
-    }
-
-    private boolean hasListener(String mapName) {
-        final EventService eventService = mapServiceContext.getNodeEngine().getEventService();
-        return eventService.hasEventRegistration(SERVICE_NAME, mapName);
-    }
+public interface EvictionOperator {
```

Comment:
Is this really an operation?? So is there an operation that implement EvictionOperator interface?